src/marten/db/query/set.cr

```diff
@@ -976,6 +976,21 @@ module Marten
           raise NotImplementedError.new("#sum is not supported for query sets")
         end
 
+        # Returns the sum of a field for the current query set.
+        #
+        # Calculates the total sum of values within the specified field for the records
+        # included in the query set. For example:
+        #
+        # ```
+        # order_items = OrderItem.filter(order_id: 123)
+        # total_price = order_items.sum(:price)
+        # ```
+        #
+        # This would calculate the total cost of all items within order number 123.
+        def sum(field : String | Symbol)
+          @query.sum(field.to_s)
+        end
```

Comment:
Can we also add specs for this method? Especially, verifying that the method can be called with both strings and symbols? 🙏

Reply:
Yes! I assume that public methods should always be tested. I always test the logic of methods in the spec associated with the corresponding abstractions, and for methods that simply "delegate" I define a simple spec verifying that we have the right behavior in place.

The comment you are mentioning needs a bit more context though! In the other PR, specs that were testing `Marten::DB::Query::SQL::Query` abstraction were essentially duplicated and redefined at an outer level for the `Marten::DB::Query::Set` abstraction. That's something we want to avoid!

But any public method should be tested anyhow. As you said, this method essentially delegates to the `Marten::DB::Query::SQL::Query` abstraction, so at least one test verifying that the method produces the expected behavior for a simple case (basic sum of a query set) is important in my opinion (like the one you added for the `Querying` concern for example). The other "logic' to test and that is specific to this method is the fact that you can call it with a field name expressed as a symbol or a string and that this produces the same result.